integration-tests/api/__tests__/store/cart/ff-sales-channels.js

```diff
@@ -0,0 +1,163 @@
+const path = require("path")
+
+const startServerWithEnvironment =
+  require("../../../../helpers/start-server-with-environment").default
+const { useApi } = require("../../../../helpers/use-api")
+const { useDb } = require("../../../../helpers/use-db")
+
+const {
+  simpleCartFactory,
+  simpleRegionFactory,
+  simpleShippingOptionFactory,
+  simpleCustomShippingOptionFactory,
+  simpleProductFactory,
+  simplePriceListFactory,
+  simpleDiscountFactory,
+} = require("../../../factories")
+const { IdMap } = require("medusa-test-utils")
+
+jest.setTimeout(30000)
+
+const customerData = {
+  email: "medusa@test.dk",
+  password: "medusatest",
+  first_name: "medusa",
+  last_name: "medusa",
+}
+
+describe("[MEDUSA_FF_SALES_CHANNELS] /store/carts", () => {
+  let medusaProcess
+  let dbConnection
+
+  beforeAll(async () => {
+    const cwd = path.resolve(path.join(__dirname, "..", "..", ".."))
+    const [process, connection] = await startServerWithEnvironment({
+      cwd,
+      env: { MEDUSA_FF_SALES_CHANNELS: true },
+      verbose: false,
+    })
+    dbConnection = connection
+    medusaProcess = process
+  })
+
+  afterAll(async () => {
+    const db = useDb()
+    await db.shutdown()
+
+    medusaProcess.kill()
+  })
+
+  describe("POST /store/carts/:id", () => {
+    let product
+
+    beforeEach(async () => {
+      await simpleRegionFactory(dbConnection, {
+        id: "test-region",
+        currency_code: "usd",
+        countries: ["us"],
+        tax_rate: 20,
+        name: "region test",
+      })
+
+      product = await simpleProductFactory(dbConnection, {
```

Comment:
```suggestion
      const product = await simpleProductFactory(dbConnection, {
```
seems to be assigned only here